spec/models/authorization_request_spec.rb

```diff
@@ -16,6 +16,23 @@
     end
   end
 
+  describe 'authorization_request_for' do
+    let!(:authorization_requests_entreprise) { create(:authorization_request, api: 'entreprise') }
+    let!(:authorization_requests_particulier) { create(:authorization_request, api: 'particulier') }
+
+    let!(:authorization_request_list) { described_class.all }
+
+    describe 'entreprise' do
+      it 'contains the enterprise authorization request' do
+        expect(authorization_request_list.authorization_requests_for('entreprise')).to include(*authorization_requests_entreprise)
+      end
+
+      it 'does not contains the particulier authorization request' do
+        expect(authorization_request_list.authorization_requests_for('entreprise')).not_to include(authorization_requests_particulier)
+      end
+    end
+  end
```

Comment:
Je ferai directement ça:

```suggestion
    subject { described_class.authorization_requests_for('entreprise') }

    it 'contains the entreprise authorization request' do
      is_expected.to include(*authorization_requests_entreprise)
    end

    it 'does not contains the particulier authorization request' do
      is_expected.not_to include(authorization_requests_particulier)
    end
  end
```

ça me semble assez straightforward, limit le texte du it on peut aussi le flip.

Reply:
après c'est good enough aussi comme ça en vrai